contrib/scalafix/src/python/pants/contrib/scalafix/tasks/scalafix.py

```diff
@@ -0,0 +1,123 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import filecmp
+import os
+import shutil
+from abc import abstractmethod
+
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.java.jar.jar_dependency import JarDependency
+from pants.option.custom_types import file_option
+from pants.util.dirutil import safe_mkdir_for
+from pants.util.meta import AbstractClass
+
+
+class ScalaFix(NailgunTask, AbstractClass):
+  """Executes the scalafix tool."""
+
+  _SCALAFIX_MAIN = 'scalafix.cli.Cli'
+  _SCALA_SOURCE_EXTENSION = '.scala'
+  _SCALAHOST_NAME = 'scalahost-nsc'
+
+  cache_target_dirs = True
+
+  @classmethod
+  def register_options(cls, register):
+    super(ScalaFix, cls).register_options(register)
+    register('--config', type=file_option, default=None, fingerprint=True,
+             help='The config file to use (in HOCON format).')
+    register('--rewrites', default=None, fingerprint=True,
+             help='The `rewrites` arg to scalafix: generally a name like `ProcedureSyntax`.')
+    cls.register_jvm_tool(register,
+                          'scalafix',
+                          classpath=[
+                            JarDependency(org='ch.epfl.scala', name='scalafix-cli_2.11.11', rev='0.4.2'),
+                          ])
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(ScalaFix, cls).prepare(options, round_manager)
+    round_manager.require_data('runtime_classpath')
+
+  @abstractmethod
+  def _finalize_target(self, target, results_dir):
+    """Given a Target and a successfully populated results_dir containing its sources, finalize."""
+
+  def _is_fixed(self, target):
+    return target.has_sources(self._SCALA_SOURCE_EXTENSION) and (not target.is_synthetic)
+
+  def execute(self):
+    classpaths = self.context.products.get_data('runtime_classpath')
+    # NB: This task uses only the literal classpath of each target, so does not need
```

Comment:
Something like "uses only the classes directly generated from the sources of the target"?